server/handlers/handlers.go

```diff
@@ -486,3 +499,18 @@ func (ah *ActivityHandler) updateLastReceivedChangeDate(t time.Time) {
 		ah.plugin.GetAPI().LogError("Unable to store properly the last received change")
 	}
 }
+
+func (ah *ActivityHandler) isActiveUser(userID string) bool {
+	mmUser, err := ah.plugin.GetAPI().GetUser(userID)
```

Comment:
Every time this function is called, we are calling the Mattermost API `GetUser` to get the user object. Check if in some of the places, we're already calling this API or not. If yes, then there's no need to call it again here.